backbone/README.md

```diff
@@ -368,3 +369,110 @@ return Marionette.ItemView.extend({
+
+The main benefit here is that you end up writing testable code, which is especailly important for JavaScript as trying to add tests later you’ll find that your code is usually not decoupled enough.
+
+Also, this means that all features have associated test cases, which is important as we strive to ship quality code.
+
+And because you fail the test first, you have more confidence when it turns green that you are testing what you think you are testing. All too often a test will pass, but you aren’t testing the correct behaviour.
+
+#### Benefits
+
+-   If you follow red/green/refactor you end up with loosely-coupled, testable code.
+-   The tests cases are readable in english, so the failure messages give you a very good idea of what actually failed- Higher code quality.
+
+### Spies, Stubs, Mocks and Fake Servers with Sinon
+
+Sinon is a testing utility that we use with our Jasmine Specs for spies, stubs and mocks. Jasmine has some similar functionality, but we have decided to Sinon’s test utilities instead.
+
+-   Spies are functions that either wrap another function or are standalone anonymous functions that you can assert have been called in some capacity.
+-   Stubs are like spies, but they have pre-programmed behaviour and if you wrap an existing function with it, the original function is not called which is often useful.
+-   Mocks are a combination of spies and stubs, but you specify the expected behaviour up front.
+-   The Fake Server allows us to test AJAX requests without a real server and supplying the different responses to test how the app responds. For the API of the Fake Server, see `fake_server.js`.
+
+I recommend reading the sinon docs for more information. <http://sinonjs.org/docs/>. Here is an example from our code using spies and the fake server:
+	
+	eb = require('eb');
+	fakeServer = require('es6!require/helpers/tests/fake_server');
+
+    it('should trigger an activated event on valid server response', function() {
+        var userIsActivatedSpy = sinon.spy();
+        this.server = eb.sinon.fakeServer.create();
+        this.user.on('activated', userIsActivatedSpy);
+        this.user.set('email', 'valid_email@example.com');
+
+        // this would normally trigger an ajax call, but it hits our fake server instead
+        this.user.fetch({
+            'url': 'get-tickets-account-status'
+        });
+        // let's return the following response from the fake server
+        fakeServer.respondWithSuccess(this.server, {
+        	user_exists: true,
+            activation_email_sent: false
+        });
+        // we can assert that our spy was called
+        expect(userIsActivatedSpy).toHaveBeenCalledOnce();
+        this.server.restore();
+    });
+
+We have extended Sinon with a few custom functions, which you can see in django/media/django/js/tests/jasmine/spec/SpecHelper.js
+
+### Running the Jasmine Tests
+
+Go to <https://www.evbdev.com/js-tests/require> to run the entire JS unit test suite.
```

Comment:
the original path (where the index of every js tests exists is `https://www.evbdev.com/js-tests/` 
require/ is not the entire, is just what lives on `js/src/require` (and excluding `react` directory under that giving that react directoy have a different runner and a different jasmine version